app/sensors/other_sensors/TPH_sensor_BME280.py

```diff
@@ -74,13 +74,19 @@ def __get_data(self):
         Returns:
             dict: A dictionary containing temperature, pressure, and humidity values.
         """
-        data = bme280.sample(self.bus, self.address, self.calibration_params)
-
-        return {
-            "temperature": round(data.temperature, 2),
-            "pressure": round(data.pressure * 0.750061, 2),
-            "humidity": round(data.humidity, 2)
-        }
+        try:
+            data = bme280.sample(self.bus, self.address, self.calibration_params)
+        except AttributeError:
+            logging.error("Attribute error.")
+        except Exception as e:
+            logging.error(e, exc_info=True)
```

Comment:
1. Please update the logging to specify which sensor caused the error. For example, start the message with `"An error occurred while reading the BME280 sensor"` and then include the error details. This will make it clear in the logs which sensor caused the error.

2. Please specify and handle different possible exceptions separately. This will allow more precise logging and easier debugging.

---

This change should be applied to the code for all sensors.